src/main/java/frc/robot/subsystems/BatterySubsystem.java

```diff
@@ -0,0 +1,199 @@
+    sbTimerChange = btTab.add("Change Timer", 0).withSize(0, 1).withPosition(4, 0).getEntry();
+    // Starts general timer
+    timer.start();
+    DriverStationSim.setSendError(true);
+  }
+
+  public void periodic() {
+    // Sets shuffleboard tabs to their respective values
+    sbVoltage.setDouble(getVoltage());
+    sbInputCurrent.setDouble(getInputCurrent());
+    sbSimVoltage.setNumber(powerDistribution.getVoltage());
+    sbTimer.setDouble(getGeneralTimer());
+    sbTimerHighCurrent.setDouble(getHighCurrentTimer());
+    sbTimerChange.setBoolean(checkTimer()); // Replace checkVoltage() with checkTimer() if necessary
+    checkCurrent();
+    checkVoltage();
+  }
+
+  public double getVoltage() {
+    return powerDistribution.getVoltage();
+  }
+
+  public double getInputCurrent() {
+    if (Robot.isSimulation()) {
+      return powerDistribution.getCurrent(1);
+    } else {
+      return powerDistribution.getTotalCurrent();
+    }
+  }
+
+  // If the current is below a certain level, stops the high current timer
+  public void checkCurrent() {
+    if (getInputCurrent() < Constants.highBatteryCurrentThreshold) {
+      stopHCTimer();
+    } else {
+      startHCTimer();
+    }
+  }
+
+  // Displays alerts based on the state of the timer
+  public boolean checkTimer() {
+    if (currentTimer.hasElapsed(Constants.timeInSecondsHighCurrentRed)) {
+      DriverStation.reportWarning("Change the Battery Now! (HCTR)", false);
+      return true;
+    } else if (timer.hasElapsed(Constants.timeInSecondsGeneralRed)) {
+      DriverStation.reportWarning("Change the Battery Now! (GTR)", false);
+      return true;
+    } else if (currentTimer.hasElapsed(Constants.timeInSecondsHighCurrentYellow)) {
+      DriverStation.reportWarning("Change the Battery Soon! (HCTY)", false);
+      return false;
+    } else if (timer.hasElapsed(Constants.timeInSecondsGeneralYellow)) {
+      DriverStation.reportWarning("Change the Battery Soon! (GTY)", false);
+      return false;
+    }
+    return false;
+  }
+
+  public double getGeneralTimer() {
+    Preferences.setDouble("Battery General Timer", timer.get() + storedGeneralTime);
+    return timer.get() + storedGeneralTime;
+  }
+
+  public double getHighCurrentTimer() {
+    Preferences.setDouble("Battery High Current Timer", currentTimer.get() + storedHighCurrentTime);
+    return currentTimer.get() + storedHighCurrentTime;
+  }
+
+  private void startHCTimer() {
+    currentTimer.start();
+  }
+
+  private void stopHCTimer() {
+    currentTimer.stop();
+  }
+
+  public void checkVoltage() {
+    if (getVoltage() < Constants.minVoltageRed) {
+      DriverStation.reportWarning("Change the Battery Now! (VR)", false);
+    } else if (getVoltage() < Constants.minVoltageYellow) {
+      DriverStation.reportWarning("Change the Battery Soon! (VY)", false);
+    } else {
+    }
+  }
+
+  public boolean checkRedVoltage() {
+    return getVoltage() > Constants.minVoltageRed;
+  }
+
+  public void resetTimers() {
```

Comment:
Good work figuring the Preferences out!
One addition I would make is to save the last 'low current' voltage as a Preference. 
Then if you read a higher voltage on startup, reset the timers because the battery has likely been changed.

Reply:
done, but still needs testing

Reply:
did some testing, seems like no matter what, if you reboot the roborio it clears the values of preferences. Either way, performance should be as intended